src/main/kotlin/com/petqua/domain/oauth/kakao/KakaoOauthClient.kt

```diff
@@ -0,0 +1,37 @@
+package com.petqua.domain.oauth.kakao
+
+import com.petqua.domain.oauth.OauthClient
+import com.petqua.domain.oauth.OauthServerType
+import com.petqua.domain.oauth.OauthServerType.KAKAO
+import com.petqua.domain.oauth.OauthTokenInfo
+import com.petqua.domain.oauth.OauthUserInfo
+import org.springframework.stereotype.Component
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+
+@Component
+class KakaoOauthClient(
+    private val kakaoOauthConfig: KakaoOauthConfig,
+    private val kakaoOauthApiClient: KakaoOauthApiClient,
+) : OauthClient {
+
+    override fun oauthServerType(): OauthServerType {
+        return KAKAO
+    }
+
+    override fun requestOauthUserInfo(code: String): OauthUserInfo {
+        val oauthTokenInfo = requestToken(code)
+        return kakaoOauthApiClient.fetchUserInfo("Bearer " + oauthTokenInfo.accessToken)
+    }
+
+    private fun requestToken(code: String): OauthTokenInfo {
+        val body: MultiValueMap<String, String> = LinkedMultiValueMap()
+        body.add("grant_type", "authorization_code")
+        body.add("client_id", kakaoOauthConfig.clientId)
+        body.add("redirect_uri", kakaoOauthConfig.redirectUri)
+        body.add("code", code)
+        body.add("client_secret", kakaoOauthConfig.clientSecret)
+
+        return kakaoOauthApiClient.fetchToken(body)
+    }
```

Comment:
하나의 메서드(`requestOauthUserInfo`)에서 서로다른 두 개의 API 호출이 이루어지는데요.

메서드를 나누는건 어떨까요? 테스트 하기에도 용이 할 것 같습니다.
OAuth2.0 플로우 내에서는 [accessToken 발급] -> [사용자 정보 요청]이 모두 동일해서, 인터페이스인 `OAuthClient`에 명세를 강력하게 선언하면 좋아보입니다.
외부에서도 별도로 호출하고요!

```suggestion
    override fun requestOauthUserInfo(tokenInfo: OauthTokenInfo): OauthUserInfo {
        return kakaoOauthApiClient.fetchUserInfo("Bearer " + tokenInfo.accessToken)
    }

    override fun requestToken(code: String): OauthTokenInfo {
        val body: MultiValueMap<String, String> = LinkedMultiValueMap()
        body.add("grant_type", "authorization_code")
        body.add("client_id", kakaoOauthConfig.clientId)
        body.add("redirect_uri", kakaoOauthConfig.redirectUri)
        body.add("code", code)
        body.add("client_secret", kakaoOauthConfig.clientSecret)

        return kakaoOauthApiClient.fetchToken(body)
    }
```

Reply:
너무 좋은데요??